### DISCUSSION TITLE
app store listener subscribed and did renew @ LIAP

### DISCUSSION THREAD

**Discussion by mretcoyy** [imdhemy/laravel-in-app-purchases#187] [Q&A] [answered]
why my event listener is not working? I already followed the documentation

**Comment by mretcoyy** [comment]
'eventListeners' => [

        DidRenew::class => [\App\Listeners\AppleDidRecoverListener::class],
        Subscribed::class => [App\Listeners\AppleSubscribe::class]
        
]


use Imdhemy\Purchases\Events\AppStore\Subscribed;

class AppleSubscribe
{
    /**
     * Create the event listener.
     *
     * @return void
     */
    public function __construct()
    {
        
    }

    /**
     * Handle the event.
     *
     * @param  object  $event
     * @return void
     */
    public function handle(Subscribed $event)
    {
        $user = [
            'name' => 'marco',
            'age' => '1',
            'prayerIntentionMsg' => 'test listener',
        ];
        $this->notifyUserAboutUpdate($user);
    }

    private function notifyUserAboutUpdate($user)
    {
        
        Mail::to('email231@gmail.com')->send(new MailPrayerIntention($user));
    }

**Comment by imdhemy** [comment, marked as answer]
Hello @mretcoyy 

- Make sure that the endpoint is accessible through the internet. Not a localhost 😄 
- Google and Apple don't send the initial purchase event, I don't know why...
- Please, refer to the following discussion to find more information on how to handle this issue. 
https://github.com/imdhemy/laravel-in-app-purchases/discussions/182

**Reply by mretcoyy** [reply]
I read the discussion #182 but I already did that in verifying receipt.
I'm on renewal of subscription in both IOS and android.

I already got the receipt via mobile application and already verified it.
I save already the information needed in the database especially the original_transaction_id

I'm on the event listener now, for cancellation and renewal of subscription in both IOS and Android.
I already done the setup below:

- End point is accessible
- End point is already set in both app store and Google Play
-I already created a test code for the event listener (I'm testing the event via mail so it will notify me in the email if the event triggers.)
- Already done with the setup in config/liap.php

Did I miss something in this setup? 
Because my server can't receive any notification from the server.

**Reply by imdhemy** [reply]
Heya @mretcoyy 

Did you try to test your server configuration with a test notification?

**For google**, you can request a test notification from the Google Console, it will appear in the logs once received.

**For Apple**:
- Make sure you are using LIAP v `1.4`
- Add a new route to receive the notification, you can add a logger there.
- Follow the steps in this PR description (Usage section) #183

**Comment by mretcoyy** [comment]
Yes I have the end point @ gcp. With domain.

**Comment by mretcoyy** [comment]
Thank you for this information
I already tested the test notification for sandbox and I receive this.

Now I'm still working on the IOS server to server notification.
and I still didnt received any notification that user is renewed.

[image: image.png]

**Comment by mretcoyy** [comment]
This is my test data.

[image: image.png]
